src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -319,7 +316,7 @@ public PossiblyConditionalState Clone()
             tempMap.Add(rootTemp, (originalInputSlot, expressionType.Type));
 
             var nodeStateMap = PooledDictionary<BoundDecisionDagNode, (PossiblyConditionalState state, bool believedReachable)>.GetInstance();
-            nodeStateMap.Add(decisionDag.RootNode, (state: initialState.Clone(), believedReachable: true));
+            nodeStateMap.Add(decisionDag.RootNode, (state: PossiblyConditionalState.Create(this), believedReachable: true));
```

Comment:
:memo: The issue was that `makeDagTempSlot` above was recording the type of the `var` pattern local into `State`, based on correct state (from `expressionType`), but we'd drop that by using `initialState` which captured a prior state.